### DISCUSSION TITLE
Configure AdguardHome for ipv6 traffic

### DISCUSSION THREAD

**Discussion by dtrannc** [AdguardTeam/AdGuardHome#5605] [Q&A] [answered]
Hi guys,

I really like Adguard Home and installed it on my Ubuntu Server 22.04 LTS according to instruction. Then I set it as DNS server for all my devices in local network (I set Adguard Home ipv4 address in router's dhcp DNS). Some traffic that uses IPV4 DNS hit the server fine but IPV6 device traffic doesn't come to it at all. Does Adguard Home accept DNS IPV6 request and how to set it up?

Adguard Home network:

```
3: enp0s31f6: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 40:8d:5c:54:38:28 brd ff:ff:ff:ff:ff:ff
    inet 192.168.68.68/22 brd 192.168.71.255 scope global enp0s31f6
       valid_lft forever preferred_lft forever
    inet6 2603:6080:700:29bb::1000/128 scope global dynamic noprefixroute
       valid_lft 85797sec preferred_lft 85797sec
```

On another machine in the same network, this works
```
dig google.com @192.168.68.68

; <<>> DiG 9.10.6 <<>> google.com @192.168.68.68
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 60041
;; flags: qr rd ra; QUERY: 1, ANSWER: 6, AUTHORITY: 0, ADDITIONAL: 1
```
but this doesn't
```
dig google.com @2603:6080:700:29bb::1000

; <<>> DiG 9.10.6 <<>> google.com @2603:6080:700:29bb::1000
;; global options: +cmd
;; connection timed out; no servers could be reached
```

**Comment by ppfeufer** [comment]
You need to set up Unique Local Addresses (ULA) for your devices and add the ULA of your AGH to the DNS of the other devices.

Example:
AGH is `fd00::2`

Running this from a device with `fd00::20`
```plaintext
» dig google.com @fd00::2

; <<>> DiG 9.16.37 <<>> google.com @fd00::2
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 48539
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;google.com.                    IN      A

;; ANSWER SECTION:
google.com.             26992   IN      A       172.217.16.78

;; Query time: 2 msec
;; SERVER: fd00::2#53(fd00::2)
;; WHEN: Fri Mar 17 20:56:35 CET 2023
;; MSG SIZE  rcvd: 55
```

**Reply by dtrannc** [reply]
How do I setup ULA for my network? My router doesn’t seem to have that option.

**Reply by ppfeufer** [reply]
Usually, it's called Router Advertisement and might have several options.

In a FritzBOX for example it looks like this:
(Home Network » Network » Network Settings » IPv6 Settings)

![image](https://user-images.githubusercontent.com/2989985/226088480-0154a8d1-b793-48f1-95b7-53205d8f19c2.png)

**Reply by dtrannc** [reply]
Thanks for screenshot. My mesh router doesn't have any configuration option for LAN ipv6. The only configuration I can set is in the screenshot. Is there another way to use AdGuard for ipv6 without ULA? Otherwise I think I'm gonna try configure my server to be the DHCP server
![IMG_2507](https://user-images.githubusercontent.com/128180368/226141619-162086ed-90d2-4b9b-9a7b-ab9cb79a2589.jpeg)
![IMG_2508](https://user-images.githubusercontent.com/128180368/226141626-742d825a-562c-4108-a871-96b4520271e2.jpeg)

**Reply by ppfeufer** [reply]
I have no idea what router that is so I can't say much about how to configure it. You can try one of the `SLAAC` configurations though.

Unfortunately, every router handles that differently.

**Reply by angorn23** [reply]
I’m having the same issue, IPv4 works great, but most of my traffic is being routed though IPv6 and so not protected.  My router is the TP Link Deco X55’s.  Any help would be very appreciated!

**Reply by dtrannc** [reply, marked as answer]
I have Deco X55 as well, and I'm able to make it work with ipv6 for my AGH on ubuntu server 22.04 LTS:

**Approach 1** - Not using AGH DHCP server:
In Deco app, under Internet Connection (IPV6) use DHCPv6 and there is a LAN section, note down the router's ipv6 address and the prefix. Now in ubuntu add below netplan config and then `sudo netplan apply`

```
network:
  version: 2
  renderer: networkd
  ethernets:
    enp0s31f6:
      dhcp6: false
      accept-ra: false
      addresses:
        - {your lan ipv6 prefix}::1/64
      routes:
        - to: default
          via: {your router's ipv6}
      nameservers:
        addresses:
          - 2606:4700:4700::1111
          - 2606:4700:4700::1001
```
Then set DNS to Manual and put {your lan ipv6 prefix}::1 as your primary DNS server
With this approach however you won't be able to reserve your IPV6 address so you can't really custom configure your clients

**Approach 2** - Configure your server to act as router then use AGH as DHCP server and turn TPlink Deco to bridge mode

**Approach 3** - Turn off IPV6 in Internet Connection

**Reply by hslabbert** [reply]
If you have a separate server running AGH (the Ubuntu box), couldn't you just use radvd to advertise a ULA with no gateway (`AdvDefaultLifetime 0`, eg ref: https://serverfault.com/a/1084476)? Then point your clients to the AGH's ULA for their DNS?
